src/RA_StringUtils.h

```diff
@@ -45,9 +48,21 @@ _NODISCARD inline auto StringPrintf(_In_z_ _Printf_format_string_ const CharT* c
     }
     else if constexpr(std::is_same_v<CharT, wchar_t>)
     {
+        // starting capacity to reduce the need to reallocate, it could still be too small but is handled below
+        sFormatted.reserve(64U);
+        nNeeded = -1; // assume it already failed (usually does)
         assert(std::wstring_view{ sFormat }.find(L"%n") == std::wstring_view::npos);
         assert(sFormatted.capacity() > 0U && (sFormatted.capacity() < RSIZE_MAX/sizeof(wchar_t)));
-        nNeeded = std::vswprintf(sFormatted.data(), sFormatted.capacity(), sFormat, pArgs);
+
+        while(nNeeded < 0) /*constant at best, logk N at worst; where k is the amount of times doubling capacity*/
+        {
+            nNeeded = std::vswprintf(sFormatted.data(), sFormatted.capacity(), sFormat, pArgs);
```

Comment:
Using `vswprintf_s `will not work, it will just automatically fail if the buffer is too small. With the test in `RA_AchievementPopup.cpp`, it had to double 3 times to 63 (kind of weird, `7->15->31->63`).

So I just put 64 as a default, but keep on doubling it below if it failed. From what I've read, `std::vsnprintf` automatically adjusts the buffer size but `std::vswprintf` does not.

Reply:
`vnsprintf` will return the size of buffer required. `vswprintf` does not, so we have to keep trying sizes until we find one that works. Once successful, the string is populated and can be returned. The "if needed>=0" logic below only needs to set the length.